runtime/jvmti/jvmtiHook.c

```diff
@@ -1018,6 +1027,11 @@ jvmtiHookClassPrepare(J9HookInterface** hook, UDATA eventNum, void* eventData, v
 	J9VMClassPrepareEvent * data = eventData;
 	J9JVMTIEnv * j9env = userData;
 	jvmtiEventClassPrepare callback = j9env->callbacks.ClassPrepare;
+	J9VMThread * currentThread = data->currentThread;
+
+	if (!shouldPostEvent(currentThread, NULL)) {
```

Comment:
Can we include this code under `#if JAVA_SPEC_VERSION >= 20`?

Reply:
Yes. Also I'm not quite sure what's the best place to return?

Reply:
Best not to return - use an if or a goto if necessary.

Reply:
Probably best to always emit the tracepoints.